tests/qubit_codes/test_surface_code.py

```diff
@@ -456,6 +455,95 @@ def test_boundary(self, surface_code):
                 assert not bound_points
 
 
+rectanguloid_code_params = it.product(
+    rng().integers(low=2, high=9, size=(8, 3)), ["primal", "dual"], ["open", "periodic"]
+)
+
+
+@pytest.fixture(scope="module", params=rectanguloid_code_params)
+def rectangular_surface_code(request):
+    """A function to define a rectangular surface code for use in this
+    module."""
+    distance, ec, boundaries = request.param
+    return SurfaceCode(distance, ec, boundaries), request.param
+
+
+class TestRectangularSurfaceCode:
+    """Test the SurfaceCode class for non-cubic lattices."""
+
+    def test_init_rectangular(self, rectangular_surface_code):
+        """Check the proper initialization of SurfaceCode."""
+        # obtaining a rectangular surface code
+        rect_sc, param = rectangular_surface_code
+        distance, ec, boundaries = param
+
+        # assert that the surface code is initialized correctly
+        assert len(rect_sc.dims) == 3, "Dimensionality of the surface code is not 3 but {}".format(
+            len(rect_sc.dims)
+        )
+        for i, x in enumerate(rect_sc.dims):
+            assert isinstance(
+                x, (int, np.integer)
+            ), "Entry {} of distance {} is not an integer but {}".format(i, rect_sc.dims, type(x))
+            assert x == distance[i], "Entry {} of distance {} is not intended distance {}".format(
+                i, rect_sc.dims, distance[i]
+            )
+        assert rect_sc.ec
+        assert rect_sc.bound_str
+        assert list(rect_sc.boundaries)
+        for ec in rect_sc.ec:
+            attributes = [
+                "stabilizers",
+                "syndrome_inds",
+                "syndrome_coords",
+                "bound_points",
+                "stab_graph",
+            ]
+            for att in attributes:
+                getattr(rect_sc, ec + "_" + att)
+            for att in ["all_syndrome_inds", "all_syndrome_coords"]:
+                getattr(rect_sc, att)
+
+    def test_rectangular_stabilizers(self, rectangular_surface_code):
+        """Check whether all stabilizers were generated as expected."""
+        rect_sc, param = rectangular_surface_code
+
+        for ec in rect_sc.ec:
+            dx, dy, dz = rect_sc.dims
+            cubes = getattr(rect_sc, ec + "_stabilizers")
+            # Check that there are a correct number of stabilizer elements
+            # depending on the boundary conditions.
+            if rect_sc.bound_str.startswith("open"):
+                if ec == "dual" and len(rect_sc.ec) == 2 and ec == "dual":
+                    assert (
+                        len(cubes) == dx * (dy - 1) * dz
+                    ), "Wrong number of dual stabilizers when boundaries open (ec 'both')"
```

Comment:
I am not sure if we should distinguish cases where `len(rect_sc.ec) ==2` and `len(rect_sc.ec) ==1`. I included it because it was done above for the cubic surface code, but I would argue that this is superfluous (or I don't understand the construction properly)

Reply:
@ilan-tz maybe it's good to discuss this in person